at_secondary/at_secondary_server/config/generate_config.py

```diff
@@ -0,0 +1,49 @@
+"""
+Generates the config.yaml from the config-base.yaml
+The default values in the configurations can be overridden by creating an
+config-<environment>.properties file
+The file accepts an optional parameter <environment>; defaults to production.
+Example:
+    Generating config.yaml for development.
+        create an application properties file as 'config-development.properties'
+        Run the generate_config.py passing 'development' as command line argument.
+            python3 generate_config.py -e development
+"""
+import argparse
+import yaml
+from jproperties import Properties
+
+# Initialize parser
+parser = argparse.ArgumentParser()
+# Adding optional argument
+parser.add_argument("-e", "--env", default='production')
+
+# Read arguments from command line
+environment = parser.parse_args().env
+
+# Read properties file
+configs = Properties()
+with open('config-' + environment + '.properties', 'rb') as read_prop:
+    configs.load(read_prop)
+
+with open("config-base.yaml", "r") as yamlfile:
+    yamlMap = yaml.load(yamlfile, Loader=yaml.FullLoader)
+    # Loop on each of the property in config properties.
+    for key in configs.properties:
+        fields = key.split('.')
+        temp = yamlMap
+        i = 0
+        # Iterate into the map until the key is fetched.
+        while i < len(fields):
+            # Condition is to check if the key in the map is fetched.
+            if i == len(fields) - 1:
+                temp[fields[i]] = configs.properties[key]
+                break
+            temp = yamlMap[fields[i]]
```

Comment:
I did not get this. Can you please explain what are we doing here for my understanding?

Reply:
Drill down to a specific key in the map and assign the value to the key.